modules/phase_field/src/postprocessors/FeatureFloodCount.C

```diff
@@ -702,12 +688,12 @@ FeatureFloodCount::flood(const DofObject * dof_object, int current_idx, FeatureD
    * If we reach this point (i.e. we haven't returned early from this routine),
    * we've found a new mesh entity that's part of a feature.
    */
-  unsigned int map_num = _single_map_mode ? 0 : current_idx;
-  processor_id_type rank = processor_id();
+  auto map_num = _single_map_mode ? decltype(current_idx)(0) : current_idx;
+  auto rank = processor_id();
 
   // New Feature (we need to create it and add it to our data structure)
   if (!feature)
-    _partial_feature_sets[rank][map_num].push_back(FeatureData(current_idx));
+    _partial_feature_sets[rank][map_num].emplace_back(FeatureData(current_idx));
```

Comment:
Clarification... I said:

> it will be rvalue assigned (moved) into the memory spot (or, if there is no move operator for this object then it will be copied into place

I just wanted to be clear that `emplace_back()` is passing the arguments through to the constructors of `FeatureData`.  So when I say "rvalue assigned" I mean that it will attempt to call a constructor that takes a `FeatureData &&` (called a "move constructor") and if one doesn't exist then it will try to call a constructor that takes `FeatureData &` (ie... copy constructor).

In this case it looks like `FeatureData` doesn't have move constructor so this line of code as it currently stands will be effectively doing the same thing `push_back()` would do...